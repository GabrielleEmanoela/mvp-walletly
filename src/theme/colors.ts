export const theme = {
  colors: {
    primary: '#4d539bff',
    secondary: '#f2f2f2ff',
    white: '#f5f5f5',
    outline: '#bdbdbd',
    background: '#ffffff',
    text: '#333333',
    placeholder: '#757575',
    danger: '#ff1744',
    success: '#00c853',
    warning: '#ffab00',
    gray: '#888888',
    grayLight: '#f0f0f0',
    black: '#000000',
    transparent: 'transparent',
  },
  modalOverlay: {
    flex: 1,
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
    alignItems: 'center',
  },
};
