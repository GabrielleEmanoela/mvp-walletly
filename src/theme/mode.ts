export const mode = {
  outlined: 'outlined',
  elevated: 'elevated',
  flat: 'flat',
};
