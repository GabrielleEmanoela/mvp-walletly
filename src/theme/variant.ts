type PaperTextVariant =
  | 'displayLarge'
  | 'displayMedium'
  | 'displaySmall'
  | 'headlineLarge'
  | 'headlineMedium'
  | 'headlineSmall'
  | 'titleLarge'
  | 'titleMedium'
  | 'titleSmall'
  | 'bodyLarge'
  | 'bodyMedium'
  | 'bodySmall'
  | 'labelLarge'
  | 'labelMedium'
  | 'labelSmall';

export const typographyVariant: Record<string, PaperTextVariant> = {
  titleMedium: 'titleMedium',
  bodyMedium: 'bodyMedium',
  headlineMedium: 'headlineMedium',
};
