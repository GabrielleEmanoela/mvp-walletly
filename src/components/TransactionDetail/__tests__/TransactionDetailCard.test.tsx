import React from 'react';
import { render } from '@testing-library/react-native';
import { TransactionDetailCard, TransactionDetailProps } from '..';
import { theme } from '@/theme/colors';

const defaultProps: TransactionDetailProps = {
  currency: 'Título',
  description: 'Descrição',
  status: 'completed',
  icon: 'cash',
  iconColor: theme.colors.primary,
  id: '',
  date: '',
  type: 'deposit',
  amount: 0,
};

describe('TransactionDetailCard', () => {
  it('should render correctly with all props', () => {
    const { getAllByText } = render(
      <TransactionDetailCard {...defaultProps} />,
    );
    expect(getAllByText(defaultProps.description).length).toBeGreaterThan(0);
  });

  it('should render correctly', () => {
    const { toJSON } = render(<TransactionDetailCard {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
