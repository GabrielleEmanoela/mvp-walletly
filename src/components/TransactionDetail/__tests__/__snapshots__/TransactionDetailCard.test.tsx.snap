// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetailCard should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#f5f5f5",
      "borderRadius": 16,
      "marginBottom": 24,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
  testID="transaction-detail-card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderRadius": 16,
        "elevation": 2,
        "flex": undefined,
        "padding": 16,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="transaction-detail-card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="transaction-detail-card"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            {
              "minHeight": 72,
            },
            {
              "color": "#333333",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 8,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "height": 40,
                "justifyContent": "center",
                "marginRight": 16,
                "width": 40,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#4d539bff",
                  "borderRadius": 20,
                  "height": 40,
                  "width": 40,
                },
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                {
                  "marginRight": 12,
                },
              ]
            }
            testID="transaction-detail-icon"
          >
            <Text
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                ]
              }
            >
              □
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  [
                    {
                      "minHeight": 30,
                      "paddingRight": 16,
                    },
                    {
                      "marginBottom": 2,
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Descrição
          </Text>
        </View>
        <View />
      </View>
      <View
        borderRadiusStyles={
          {
            "borderRadius": 16,
          }
        }
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            {
              "paddingBottom": 16,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "color": "#333333",
                "fontSize": 14,
                "marginBottom": 8,
              },
            ]
          }
          testID="transaction-detail-description"
        >
          Descrição
        </Text>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "color": "#00c853",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 8,
              },
            ]
          }
          testID="transaction-detail-amount"
        >
          R$ 0,00
        </Text>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "color": "#333333",
                "fontSize": 14,
                "marginTop": 4,
              },
            ]
          }
          testID="transaction-detail-currency"
        >
          Título
        </Text>
      </View>
    </View>
  </View>
</View>
`;
